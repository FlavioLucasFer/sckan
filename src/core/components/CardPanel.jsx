import React, { useEffect, forwardRef } from 'react';
import PropTypes from 'prop-types';
import { uniqueId } from 'lodash';

const elemId = uniqueId('card-panel-');

CardPanel.propTypes = {
	id: PropTypes.string,
	backgroundColor: PropTypes.string,
	className: PropTypes.string,
	cardClassName: PropTypes.string,
	noCol: PropTypes.bool,
	tooltipPosition: PropTypes.string,
	tooltip: PropTypes.string,
	onClick: PropTypes.func,
	cardStyle: PropTypes.object,
	children: PropTypes.node,
};

function CardPanel(props, ref) {
	const {
		id,
		backgroundColor,
		className,
		cardClassName,
		noCol,
		tooltipPosition,
		tooltip,
		onClick,
		cardStyle,
		children,
		...rest
	} = props;

	useEffect(() => {
		M.Tooltip.init(document.querySelectorAll('.tooltipped'));
	}, []);

	function handleClick() {
		const elems = document.querySelectorAll('.tooltipped');

		for (let i = 0; i < elems.length; i++) {
			const tooltipInstance = M.Tooltip.getInstance(elems[i]);
			tooltipInstance.close();
		}

		if (onClick) {
			onClick();
		}
	}

	return (
		<div {...rest}
			ref={ref}
			className={`${!noCol && 'col'} ${className || ''}`}>
			<div id={id || elemId}
				className={`
					card-panel 
					${tooltip && 'tooltipped'}
					${cardClassName || ''} 
					${backgroundColor || ''}
				`}
				data-position={tooltipPosition || 'bottom'}
				data-tooltip={tooltip || ''}
				style={cardStyle}
				onClick={handleClick}>
				{children}
			</div>
		</div>
	);
}

export default forwardRef(CardPanel);