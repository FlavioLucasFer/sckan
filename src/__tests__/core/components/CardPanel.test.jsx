import React from 'react';
import { shallow } from 'enzyme';

import CardPanel from 'core/components/CardPanel';

describe('Testing CardPanel core component', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<CardPanel>
				<h2>testing</h2>
			</CardPanel>
		);

		expect(wrapper.debug()).toMatchSnapshot();

		wrapper.setProps({
			backgroundColor: 'teal',
			className: 'col s12',
			cardClassName: 'col s12',
			tooltipPosition: 'right',
			tooltip: 'testing',
			onClick: () => console.log('testing'),
		});

		expect(wrapper.debug()).toMatchSnapshot();
	});
});