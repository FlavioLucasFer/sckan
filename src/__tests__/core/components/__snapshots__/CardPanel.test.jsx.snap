// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CardPanel core component should render correctly 1`] = `
"<div className=\\"col \\">
  <div id=\\"card-panel-1\\" className=\\"\\\\n\\\\t\\\\t\\\\t\\\\t\\\\tcard-panel \\\\n\\\\t\\\\t\\\\t\\\\t\\\\tundefined\\\\n\\\\t\\\\t\\\\t\\\\t\\\\t \\\\n\\\\t\\\\t\\\\t\\\\t\\\\t\\\\n\\\\t\\\\t\\\\t\\\\t\\" data-position=\\"bottom\\" data-tooltip=\\"\\" onClick={[Function: handleClick]}>
    <h2>
      testing
    </h2>
  </div>
</div>"
`;

exports[`Testing CardPanel core component should render correctly 2`] = `
"<div className=\\"col col s12\\">
  <div id=\\"card-panel-1\\" className=\\"\\\\n\\\\t\\\\t\\\\t\\\\t\\\\tcard-panel \\\\n\\\\t\\\\t\\\\t\\\\t\\\\ttooltipped\\\\n\\\\t\\\\t\\\\t\\\\t\\\\tcol s12 \\\\n\\\\t\\\\t\\\\t\\\\t\\\\tteal\\\\n\\\\t\\\\t\\\\t\\\\t\\" data-position=\\"right\\" data-tooltip=\\"testing\\" onClick={[Function: handleClick]}>
    <h2>
      testing
    </h2>
  </div>
</div>"
`;
